data.sql

```diff
@@ -41,4 +41,231 @@ INSERT INTO
+UPDATE
+    animals
+SET
+    species_id = (
+        SELECT
+            id
+        FROM
+            species
+        WHERE
+            name = 'Digimon'
+    )
+WHERE
+    Lower(name) LIKE '%mon';
+
+UPDATE
+    animals
+SET
+    species_id = (
+        SELECT
+            id
+        FROM
+            species
+        WHERE
+            name = 'Pokemon'
+    )
+WHERE
+    species_id IS NULL;
+
+-- Modify your inserted animals to include owner information (owner_id):
+-- Sam Smith owns Agumon.
+-- Jennifer Orwell owns Gabumon and Pikachu.
+-- Bob owns Devimon and Plantmon.
+-- Melody Pond owns Charmander, Squirtle, and Blossom.
+-- Dean Winchester owns Angemon and Boarmon.
+UPDATE
+    animals
+SET
+    owner_id = (
+        CASE
+            WHEN name = 'Agumon' THEN (
+                SELECT
+                    id
+                FROM
+                    owners
+                WHERE
+                    full_name = 'Sam Smith'
+            )
+            WHEN name IN ('Gabumon', 'Pikachu') THEN (
+                SELECT
+                    id
+                FROM
+                    owners
+                WHERE
+                    full_name = 'Jennifer Orwell'
+            )
+            WHEN name IN ('Devimon', 'Plantmon') THEN (
+                SELECT
+                    id
+                FROM
+                    owners
+                WHERE
+                    full_name = 'Bob'
+            )
+            WHEN name IN ('Charmander', 'Squirtle', 'Blossom') THEN (
+                SELECT
+                    id
+                FROM
+                    owners
+                WHERE
+                    full_name = 'Melody Pond'
+            )
+            WHEN name IN ('Angemon', 'Boarmon') THEN (
+                SELECT
+                    id
+                FROM
+                    owners
+                WHERE
+                    full_name = 'Dean Winchester'
+            )
+        END
+    );
+
+---------------------------------------------------------------
+--------------------------------------------------------------
+-- What animals belong to Melody Pond?
+SELECT
+    owners.full_name AS owner,
+    animals.name AS animal
+FROM
+    animals
+    JOIN owners ON animals.owner_id = owners.id
+WHERE
+    owners.full_name = 'Melody Pond';
+
+-- List of all animals that are pokemon (their type is Pokemon).
+SELECT
+    animals.name AS animal_name,
+    species.name AS species_name
+FROM
+    animals
+    JOIN species ON animals.species_id = species.id
+WHERE
+    species.name = 'Pokemon';
+-- List all owners and their animals, remember to include those that don't own any animal.
+SELECT owners.full_name AS owner, animals.name AS animal
+FROM owners
+LEFT JOIN animals ON owners.id = animals.owner_id;
+-- How many animals are there per species?
+-- List all Digimon owned by Jennifer Orwell.
+-- List all animals owned by Dean Winchester that haven't tried to escape.
+-- Who owns the most animals?
```

Comment:
- [ ][OPTIONAL] this should be in your queries.sql, and as you already have them in your queries, kindly remove them here

Reply:
Thanks a lot sir. I'm working on the suggestions ASAP.
Blessing..